source/contribute.rst

```diff
@@ -41,128 +41,170 @@ contributing back to a Doctrine project.
-always reflects the latest version. Each released stable version will be
-a tagged commit in a **doctrine/\*** branch. Each released unstable
-version will be a tagged commit in the **doctrine/2.11.x** branch.
+Have you made your choice? Good. You now need to create a topic branch.
 
-    **NOTE** You should never commit to your forked default branch (**origin/2.11.x**).
-    Changes to **origin/2.11.x** will never be merged into
-    **doctrine/2.11.x**. All work must be done in a **topic branch**,
-    which are explained below.
+Creating a topic branch
+-----------------------
 
-Topic Branches
---------------
+You should avoid making changes directly on the patch, minor or major
+release branches, also in your fork of the repository. That makes it
+difficult for you to have more than one pull request at a time, and also
+complicates matters should you need to rebase your work to another base
+branch some time later on. Instead, for every change that you would like
+to propose, create a dedicated topic branch.
 
-Topic branches are for contributors to develop bug fixes, new features,
-etc. so that they can be easily merged to **2.11.x**. They must follow a
-few rules as listed below:
+.. note::
 
--  May branch off from: **2.11.x** whenever possible, or a newer version
-   branch otherwise. Keep in mind that your changes will be
-   up-merged to higher version branches by maintainers after the merge if
-   they are applicable.
--  Branch naming convention: anything except master, the default branch name,
-   or version branch names.
+    If that does not sound all too familiar, you might want to read
+    about `GitHub Flow
+    <https://docs.github.com/en/get-started/quickstart/github-flow.>`_
+    for additional background information
 
-Topic branches are used to develop new features and fix reported issues.
-When starting development of a feature, the target release in which this
-feature will be incorporated may well be unknown. The essence of a topic
-branch is that it exists as long as the feature is in development, but
-will eventually be merged into **2.11.x** or a release branch (to
-add the new feature or bugfix to a next release) or discarded (in case
-of a disappointing experiment).
 
-Topic branches should exist in your **local** and **origin**
-repositories only, there is no need for them to exist in **doctrine**.
+Time might have elapsed since the last time you contributed or since you
+cloned the repository. You might want to fetch the latest changes from
+the ``doctrine`` remote:
 
-Creating a topic branch
------------------------
+.. code-block:: console
+
+    $ git fetch doctrine
 
-First create an appropriately named branch. When starting work on a new
-topic, branch off from **doctrine/2.11.x** or a **doctrine/\*** branch:
+.. note::
+
+   Here, we assume ``doctrine`` points to the upstream repository, and
+   not your fork. If you are not sure, you can check with ``git remote -v``.
```

Comment:
We don't need this note since we've added the remote name in the beginning section.